### PR TITLE
mailer package

### PR DESCRIPTION
**Pull request by ferpessina**

### CODE REVIEW

File: mailer.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+const nodemailer = require('nodemailer');
```
Comment (maxiruani) [inline mailer.js]: Add `'use strict';`

File: mailer.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    /**
+     * @param {Object}  options
+     * @param {String}  options.host
+     * @param {Number}  options.port
+     * @param {Boolean} options.secure
+     * @param {Object}  options.auth
+     * @param {String}  options.auth.user
+     * @param {String}  options.auth.password
+     */
+    static createInstance(options) {
```
Comment (maxiruani) [inline mailer.js]: Add `@return {Mailer}` to jsdoc

File: mailer.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        });
+    }
+
+    /**
+     * @param from
+     * @param to
+     * @param subject
+     * @param text
+     * @param html
+     * @return {*}
```
Comment (maxiruani) [inline mailer.js]: Can you improve the jsdoc for this? Thanks :)

File: mailer.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+     * @param {String}  options.host
+     * @param {Number}  options.port
+     * @param {Boolean} options.secure
+     * @param {Object}  options.auth
+     * @param {String}  options.auth.user
+     * @param {String}  options.auth.password
+     */
+    static createInstance(options) {
+        if(_instance != null) {
+            throw new Error('mailer.js::createInstance:: ERROR: Instance already created, use getInstance() instead.');
```
Comment (maxiruani) [inline mailer.js]: Improve this to:
`throw new Exception(ErrorCode.ERROR_INSTANCE_ALREADY_CREATED, 'Mailer instance is already created and initialized. Use .getInstance()')`

If you can, add those ErrorCode's to `Exception.ErrorCode`

File: mailer.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        }
+        _instance = new Mailer(options);
+    }
+
+    /**
+     * @return {Mailer}
+     */
+    static getInstance() {
+        if(_instance == null) {
+            throw new Error('mailer.js::getInstance:: ERROR: Must call createInstace() before using.');
```
Comment (maxiruani) [inline mailer.js]: Improve this to:
`throw new Exception(ErrorCode.ERROR_INSTANCE_NOT_CREATED, 'Mailer instance should be created with .createInstance()');`

File: mailer.js (hunk 2 above)
Comment (ferpessina) [reply mailer.js]: createInstance() does not return the instance it will only initialize it. To fetch the instance, getInstance() should be called.

**Pull request merged**